### PR TITLE
[Specifications] Add the remaining TCB specifications to the repository.

### PR DESCRIPTION
**Pull request by JoshLind**
## Motivation

This PR adds the remaining TCB specifications to the repository: the overview, the key manager, secure storage and safety rules. These specifications are still in a *draft* state, as there are a number of improvements/fixes still to be made. However, they're good enough to be made public for now. We should continue to iterate on them.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Rendering all files locally seems to work.

## Related PRs

None.

### CODE REVIEW

File: specifications/trusted_computing_base/safety_rules.md [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+LSR updates the persistent storage `preferred_round` to the maximum of the `quorum_cert`'s parent's round or the current stored `preferred_round`.
+
+#### Property 3
+
+> A validator can only vote on a proposal if the output of that proposal extends its parent's output.
+
+Each `Vote` contains the executed state output, the resulting root hash of the blockchain ledger after executing the proposal. In order to verify that this extends the parent block, a `VoteProposal` contains an `AccumulatorExtensionProof` which is a merkle-proof showing how the current state can be derived from the parent's state, hence ensuring that the ledger is append-only.
+
+#### Property 5
```
Comment (mimoo) [inline specifications/trusted_computing_base/safety_rules.md:190]: a quick glance tells me that there is a missing prop 4 here

Comment (davidiw) [reply specifications/trusted_computing_base/safety_rules.md:190]: weird, did you just copy and paste this @JoshLind ?

Comment (JoshLind) [reply specifications/trusted_computing_base/safety_rules.md:190]: Yeah, I literally just copied it across. Seems like a problem with the original: https://github.com/novifinancial/libra-specs/blob/master/trusted_computing_base/safety_rules.md

**Pull request closed without merging**